HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -21,6 +21,41 @@ class OAuthAccessTokenMockStore: OAuthAccessTokenStore {
     }
 }
 
+class OAuthAccessTokenInterceptorParser: OAuthAccessTokenParser {
+    
+    private let defaultParser = OAuthAccessTokenDefaultParser()
+    
+    var parseAccessTokenCalled: Bool = false
+    
+    private var shouldIntercept: Bool = false
+    private var interceptToken: OAuthAccessToken?
+    
+    func intercept(parsedToken token: OAuthAccessToken?) {
+        shouldIntercept = true
+        interceptToken = token
+    }
+    
+    func reset() {
+        parseAccessTokenCalled = false
+        shouldIntercept = false
+    }
+    
+    func parse(data: NSData) -> ParsedToken {
```

Comment:
I get the intention and agree that exposing `Result` is not the best way to go. For consistency reasons I would still use `Result` here and maybe drop it completely in future though.